discovery/gossiper.go

```diff
@@ -904,9 +903,7 @@ func (d *AuthenticatedGossiper) networkHandler() {
 
 	// We'll use this validation to ensure that we process jobs in their
 	// dependency order during parallel validation.
-	validationBarrier := routing.NewValidationBarrier(
-		runtime.NumCPU()*4, d.quit,
-	)
+	validationBarrier := routing.NewValidationBarrier(1000, d.quit)
```

Comment:
it's just meant to high enough to facilitate processing of many concurrent requests. the actual number a node may be able to do probably depends more on hardware than anything. we can't make it unlimited since it's a semaphore, and it will be a DOS vector. we'd end up reading all of the incoming gossip messages at once and performing all of the dependent operations/requests at once.